app/controllers/omniauth_controller.rb

```diff
@@ -13,7 +13,9 @@ def tra_openid_connect
       feature_flag_id: session["feature_flag_id"],
     )
 
-    if @user.persisted?
+    # @user.persisted? checks that it exists and has been persisted to the database
+    # @user.save checks that any changes made to an existing record have been persisted to that persisted record
+    if @user.persisted? && @user.save
       Services::Feature.enroll_user_in_get_an_identity_pilot(@user)
       session["user_id"] = @user.id
 
```

Comment:
minor but the `@user.errors.to_hash` on line 25 gives us enough information if the `.save` fails right? it includes all the messages?

Reply:
Sorry GH has put the line analysis here

Reply:
Actually i guess its more how sentry parses the error which is just passed as the message, il assume this is ok as its mostly working like that in other places.

Side note seems weird we have a sentry related method in this class and not in its own class that is used elsewhere around the code.

Reply:
Yes, but it shouldn't include the actual user details. Only messages like: "Email has already been taken".

Reply:
Good point redact that user data!
![](https://media.giphy.com/media/12iWbvBFuiCLUYG70b/giphy.gif)